pylint/extensions/docparams.py

```diff
@@ -309,14 +309,24 @@ def visit_raise(self, node: nodes.Raise) -> None:
         doc = utils.docstringify(func_node.doc, self.config.default_docstring_type)
         if not doc.is_valid():
             if doc.doc:
-                self._handle_no_raise_doc(expected_excs, func_node)
+                missing = {exc.name for exc in expected_excs}
+                self._handle_no_raise_doc(missing, func_node)
             return
 
         found_excs_full_names = doc.exceptions()
-
         # Extract just the class name, e.g. "error" from "re.error"
         found_excs_class_names = {exc.split(".")[-1] for exc in found_excs_full_names}
-        missing_excs = expected_excs - found_excs_class_names
+
+        missing_excs = set()
+        for expected in expected_excs:
```

Comment:
Let's use `itertools.zip_longest` here, see [this](https://stackoverflow.com/questions/11318977/zipping-lists-of-unequal-size)

Reply:
Something based on:
```suggestion
        for expected, found_exc in itertools.zip_longest(expected_excs, found_excs_class_names)
```

This allows removing an indent with the additional `for` loop.